### ISSUE TITLE
JavaScript, function parameter names get overwritten by variable names

### ISSUE THREAD

**Issue by Alexander-Zierhut** [tdewolff/minify#334]
Original file: https://pastebin.com/XWEnF5gt
Minified Beautified version: https://pastebin.com/gRv3ufMe

Error: `VM129:XXX Uncaught SyntaxError: Identifier 'a' has already been declared`

The exact part I think this issue is occurring as `action` is overwritten by `dat`. If I change the names by hand, the same error occurs in later parts of the file.

```JS
    action(action, data, handler) {
      $.ajax({
        method: "POST",
        data: Object.assign(data, {action: action})
      }).done((data) => {
        var dat = null;
        try {
          dat = JSON.parse(data);
        } catch (e) {
          console.error("Please show this to a developer: ", data);
        }
        if (dat != null) {
          handler(dat);
        }
      });
    }
```

**Comment by tdewolff** [comment]
Thanks for the bug report, this is the smallest I could make to reproduce the bug:

```
(e) => {
  for (let dataset of x) {
    if (dataset.includes(e)) {
      var item;
    }
  }
}
```

I'll work on a fix.

**Comment by tdewolff** [comment]
I should bring out a new release fairly soon, thanks again.

**Comment by Alexander-Zierhut** [comment]
You are the best!

**Comment by tdewolff** [comment]
Has been released in v2.9.6!

**Comment by Alexander-Zierhut** [comment]
Perfect!
